File: backend/src/main/java/be/lpbconsult/befintax/wallet/dto/AssetCreateDTO.java
package be.lpbconsult.befintax.wallet.dto;

import be.lpbconsult.befintax.wallet.enums.AssetType;

import java.math.BigDecimal;
import java.util.List;

public record AssetCreateDTO(
        String name,
        String symbol,
        AssetType type,
        Boolean taxCollectedByBroker,
        BigDecimal stockTaxRate,
        BigDecimal priceEnd2025,
        String currency,
        String exchange,
        List<AssetTransactionCreateDTO> transactions,
        Long walletId
) {}

File: frontend/angular/src/app/features/wallet/dtos/asset-create.dto.ts
import { AssetType } from '../../../core/models/asset.model';
import {AssetTransactionCreateDto} from './asset-transaction-create.dto';

export interface AssetCreateDto {
  name: string;
  symbol: string;
  type: AssetType;
  taxCollectedByBroker: boolean;
  stockTaxRate: number;
  priceEnd2025: number;
  currency: string;
  exchange: string;
  transactions?: AssetTransactionCreateDto[];
  walletId?: number;
}
